src/syntax/syntax.rkt

```diff
@@ -547,6 +547,17 @@
   [fl or-fn] [bf or-fn] [ival ival-or]
   [nonffi or-fn])
 
+;; Miscellaneous operators ;;
+
+(define-operator (cast cast.f64 binary64) binary64
+  [fl identity] [bf identity] [ival #f]
+  [nonffi (curry repr->real (get-representation 'binary64))])
+
+(define-operator (cast cast.f32 binary32) binary32
+  [fl (curry repr->real (get-representation 'binary32))]
```

Comment:
Why isn't the `fl` version just `identity`?

Reply:
It was actually supposed to be `real->repr`, but not the point... I think casting should force any value into a given repr, in this case, binary64 or binary32. `real->repr` is a little bit of a misnomer since it has the ability to convert a value in _any_ repr to a given repr via bigfloat.

Reply:
I'm a little unsure on the nonffi implementation. That one might be `identity`.

Reply:
Yeah, `nonffi` should be the identity, though we need to rework that anyway. Ok, then we think of casts as a special built-in operation? Why not have a `cast.posit8` that casts anything to a posit8? I think that's a bad idea: Herbie should produce programs users can run; what if they don't have a certain cast on their machine?

Reply:
Huh. Good point. I suppose we'll figure this all out when we actually implement casts and annotations. I'll probably just do `identity` for `binary64` and `real->single-flonum` for `binary32`.

Reply:
It should be `identity` for both. The `nonffi` version should really be renamed `exact?`: it's a function that computes `exact?`, `real?` outputs for `exact?`, `real?` inputs.

Reply:
Ok, good. I actually changed my mind and made it `identity` for `binary32`.